src/vivarium_testing_utils/automated_validation/data_transformation/measures.py

```diff
@@ -238,13 +257,110 @@ class RiskExposure(RatioMeasure):
     """
 
     def __init__(self, risk_factor: str) -> None:
-        self.measure_key = f"risk_factor.{risk_factor}.exposure"
+        self.measure_key = self.artifact_key = f"risk_factor.{risk_factor}.exposure"
         self.risk_factor = risk_factor
 
         # Create custom formatters for risk exposure
         self.numerator = RiskStatePersonTime(risk_factor)
         self.denominator = RiskStatePersonTime(risk_factor, sum_all=True)
 
+    @check_io(artifact_data=SingleNumericColumn, out=SingleNumericColumn)
+    def get_measure_data_from_artifact(self, artifact_data: pd.DataFrame) -> pd.DataFrame:
+        return artifact_data
+
+
+class CategoricalRelativeRisk(RatioMeasure):
+    """Computes relative risk of a categorical variable."""
+
+    def __init__(
+        self,
+        risk_factor: str,
+        affected_entity: str,
+        affected_measure: str,
```

Comment:
I think this is what you would need to just pass to RatioMeasure and used the mapping to infer measure and artifact key.